packages/validator/src/cli-validator/run-validator.js

```diff
@@ -246,9 +249,14 @@ const processInput = async function(program) {
         chalk
       );
 
-      process.chdir(path.dirname(validFile));
-      // let spectral handle the parsing of the original swagger/oa3 document
-      spectralResults = await spectral.run(originalFile);
+      const fileExtension = ext.getFileExtension(validFile);
+      let parser = Parsers.Json;
+      if (['yaml', 'yml'].includes(fileExtension)) {
```

Comment:
I guess you weren't appalled by the way I did this. 😂

Reply:
Haha nope! Maybe not the exact same as I would've done but functionally the same. And this code was already there 🙂